src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/StaticHolderTypes.cs

```diff
@@ -73,13 +73,18 @@ public override void Initialize(AnalysisContext context)
         private static void AnalyzeSymbol(SymbolAnalysisContext context)
         {
             var symbol = (INamedTypeSymbol)context.Symbol;
+
             if (!symbol.IsStatic &&
-                symbol.MatchesConfiguredVisibility(context.Options, Rule, context.Compilation, context.CancellationToken) &&
+                !symbol.IsAbstract &&
+                !IsSealedAndVisualBasic(symbol) &&
```

Comment:
In vb.net the rule recommends to mark the class as "sealed" (`NotInheritable`) because `static` doesn't exist, so if the class is already marked as such, bail-out before doing more complex process. Note that the change is required because before the `IsStaticHolderType` was returning false for a vbnet static-like type holder causing `CA1812` to report FPs.